### DISCUSSION TITLE
Unable to build debian package on Debian bookworm (arm64)

### DISCUSSION THREAD

**Discussion by solskogen** [86Box/86Box#3858] [Q&A] [answered]
### What happened?

```
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LOCALSTATEDIR
    CMAKE_INSTALL_RUNSTATEDIR
    CMAKE_INSTALL_SYSCONFDIR
    FETCHCONTENT_FULLY_DISCONNECTED


-- Build files have been written to: /home/solskogen/fjas/86Box/build/regular
make[1]: Leaving directory '/home/solskogen/fjas/86Box'
   dh_auto_build -a -O--buildsystem=cmake\+ninja
        cd obj-aarch64-linux-gnu && LC_ALL=C.UTF-8 ninja -j4 -v
ninja: error: loading 'build.ninja': No such file or directory
dh_auto_build: error: cd obj-aarch64-linux-gnu && LC_ALL=C.UTF-8 ninja -j4 -v returned exit code 1
make: *** [debian/rules:25: binary-arch] Error 1
dpkg-buildpackage: error: debian/rules binary-arch subprocess returned exit status 2


```
### Configuration file

```ini
N/A
```

### Operating system

Debian bookworm

### CPU

Raspberry Pi

### 86Box version

git/master

### Build architecture

Linux - ARM (64-bit)

### Build type

- [ ] New recompiler
- [ ] Debug build

### Download source

Official website (Jenkins, GitHub)

**Comment by cold-brewed** [comment]
Please provide the full command used for the package build.

**Comment by solskogen** [comment]
`dpkg-buildpackage -us -uc -B`

**Comment by cold-brewed** [comment, marked as answer]
I was able to reproduce this on  both x86_64 and aarch64 bookworm.

Give this a try:

In `debian/rules`, change this line:
```
dh_auto_configure --buildsystem cmake+ninja -- --preset regular --toolchain $(TOOLCHAIN) -DNEW_DYNAREC=$(NDR)
```
to
```
dh_auto_configure --buildsystem cmake+ninja -- --preset regular --toolchain $(TOOLCHAIN) -DNEW_DYNAREC=$(NDR) -B .
```

This is basically just adding `-B .` to specify the build directory. The debian cmake support seems to want it in the root. This seemed to fix the issue for me.

Note: If you are trying to run on aarch64 linux you should use our official AppImage available either in the releases section here on GitHub or [from our CI](https://ci.86box.net/job/86Box/). The debian package building is provided on a best-effort basis. The AppImage builds we provide ensure that all the necessary dependencies are included.

**Comment by solskogen** [comment]
Thank you that seems to work.
It's also missing some dependencies:

I also had to install: `libfluidsynth-dev libxkbcommon-x11-dev extra-cmake-modules cmake-extras` for cmake to be happy.

**Reply by actraiser30** [reply]
> dev

Hi, what are the total packages that you install in order to build it?

**Comment by solskogen** [comment]
and `qtbase5-private-dev` (which cmake doesn't search for)

**Comment by OBattler** [comment]
Converting this to a discussion since it has to be discussed what to do with this wrt. long-term maintenance.

**Reply by jriwanek** [reply]
I'm quite happy to maintain it, and as such, I've opened a PR with forward progress.

**Reply by actraiser30** [reply]
It does not work on buster and bullseye what makes you think will work in bookworm?

Useless
